src/components/AlphabetFilter/AlphabetFilter.tsx

```diff
@@ -0,0 +1,197 @@
+      { text: "O", value: "o" },
+      { text: "P", value: "p" },
+      { text: "Q", value: "q" },
+      { text: "R", value: "r" },
+      { text: "S", value: "s" },
+      { text: "T", value: "t" },
+      { text: "U", value: "u" },
+      { text: "V", value: "v" },
+      { text: "W", value: "w" },
+      { text: "X", value: "x" },
+      { text: "Y", value: "y" },
+      { text: "Z", value: "z" },
+      { text: "Show All", value: "showAll" },
+    ];
+
+    const refCurrentLetter = useRef(currentLetter);
+    const [selectedLetter, setSelectedLetter] = useState<string>(currentLetter);
+
+    // If the parent passes down a new currentLetter, and an onClick
+    // function exists, then set the internal state – selectedLetter –
+    // to the new currentLetter and update the refCurrentLetter with that value.
+    React.useEffect(() => {
+      if (
+        onClick &&
+        currentLetter &&
+        currentLetter !== refCurrentLetter.current
+      ) {
+        setSelectedLetter(currentLetter);
+        refCurrentLetter.current = currentLetter;
+      }
+    }, [currentLetter, onClick]);
+
+    /**
+     * This function is only called when clicking on a link should
+     * not update the URL or refresh the page.
+     */
+    const handleOnClick = (
+      e: React.MouseEvent | React.KeyboardEvent,
+      clickedLetter: string
+    ) => {
+      e.preventDefault && e.preventDefault();
+      if (selectedLetter === clickedLetter) return;
+      setSelectedLetter(clickedLetter);
+      onClick && onClick(clickedLetter);
+    };
+
+    const getButtonElement = (item) => {
+      const isSelectedLetter =
+        selectedLetter === item.value && item.value !== "showAll";
+      const isShowAll = item.value === "showAll";
+      const isButtonDisabled =
+        isDisabled ||
+        (activeLetters && !activeLetters.includes(item.value) && !isShowAll);
+
+      const buttonStyles = isSelectedLetter
+        ? { ...styles, border: "1px solid gray" }
+        : isShowAll
+        ? {
+            ...styles,
+            fontWeight: "normal !important",
+            whiteSpace: "nowrap",
+            marginLeft: "var(--nypl-space-m)",
+          }
+        : styles;
+
+      return (
+        <Button
+          aria-current={isSelectedLetter ? "page" : null}
+          aria-label={
+            item.text === "Show All" ? item.text : "Page " + item.text
+          }
+          buttonType="link"
+          id={`filter-${item.value}`}
+          isDisabled={isButtonDisabled}
+          __css={buttonStyles}
+          onClick={(e: React.MouseEvent | React.KeyboardEvent) => {
+            handleOnClick(e, item.value);
+          }}
+        >
+          {item.text}
+        </Button>
+      );
+    };
+
+    const getFilterLetters = () => {
+      if (isLargerThanMobile) {
+        return filterButtons.map((item) => {
+          return getButtonElement(item);
+        });
+      }
+      // split letters into rows of 10 + Show All
```

Comment:
@bigfishdesign13  The length of "10" is not defined in the spec doc but we (@Toxiapo and I) are assuming that this is because the Figma design is broken up so that there are 10 letters on each row. Is this intentional or should it be dynamic based on the width of the viewport, e.g. should a slightly larger mobile phone display 12 letters rather than 10 per row (should it wrap). Can you make this clearer in the spec doc as well?

Reply:
To bring this component into better alignment with how we handle buttons for mobile, I updated the component in the Visual Design Library (the Figma Main file).  For the mobile view, the buttons now used fixed dimensions of `44 x 44`, which actually reduces the number of buttons that would fit in the design.  With that, let’s make it so the buttons are in one flex row and allow them to wrap onto new lines as needed based on the width of the device.

I will update the spec doc to reflect this.

Reply:
The spec doc has been updated…
https://docs.google.com/document/d/13IOduQYuHW4t_1EV05dJVfzt_d-FTNkyczZmHCgJEO4/edit#heading=h.kuqwpe6igp9d

Reply:
Updated to Flex with specified dimensions